src/basic/utf8.c

```diff
@@ -196,42 +196,86 @@ char *utf8_escape_invalid(const char *str) {
         return p;
 }
 
-char *utf8_escape_non_printable(const char *str) {
-        char *p, *s;
+static int utf8_char_console_width(const char *str) {
+        char32_t c;
+
+        if (utf8_encoded_to_unichar(str, &c) < 0)
+                return -1;
```

Comment:
why not propagate the original error here (i.e. -EINVAL)?

Reply:
I just moved that code around. But I agree, propagating is better. Will fix.